.clang-format

```diff
@@ -0,0 +1,11 @@
+---
+BasedOnStyle: LLVM
+IndentWidth: 4
+ColumnLimit: 120
+
+AllowShortFunctionsOnASingleLine: false
+
+BreakBeforeBraces: Custom
+BraceWrapping:
+  AfterFunction: true
+  AfterStruct: true
```

Comment:
Just start to try clang-format, not well know all configuration. Current configuration seems not work well.
Some defines change:
```c
/* interrupt related */
#define ETS_SPI_INUM 2
#define ETS_GPIO_INUM 4
#define ETS_UART_INUM 5
#define ETS_MAX_INUM 6
#define ETS_SOFT_INUM 7
#define ETS_WDT_INUM 8
#define ETS_FRC_TIMER1_INUM 9
```
this can be the old style?

```c
#define GPIO_PIN_REG(i)                                                                                             \
    (i == 0)                                                                                                        \
        ? GPIO_PIN_REG_0                                                                                            \
        : (i == 1)                                                                                                  \
              ? GPIO_PIN_REG_1                                                                                      \
              : (i == 2)                                                                                            \
                    ? GPIO_PIN_REG_2                                                                                \
                    : (i == 3)                                                                                      \
                          ? GPIO_PIN_REG_3                                                                          \
                          : (i == 4)                                                                                \
                                ? GPIO_PIN_REG_4                                                                    \
                                : (i == 5)                                                                          \
                                      ? GPIO_PIN_REG_5                                                              \
                                      : (i == 6)                                                                    \
                                            ? GPIO_PIN_REG_6                                                        \
                                            : (i == 7)                                                              \
                                                  ? GPIO_PIN_REG_7                                                  \
                                                  : (i == 8)                                                        \
                                                        ? GPIO_PIN_REG_8                                            \
                                                        : (i == 9)                                                  \
                                                              ? GPIO_PIN_REG_9                                      \
                                                              : (i == 10)                                           \
                                                                    ? GPIO_PIN_REG_10                               \
                                                                    : (i == 11)                                     \
                                                                          ? GPIO_PIN_REG_11                         \
                                                                          : (i == 12)                               \
                                                                                ? GPIO_PIN_REG_12                   \
                                                                                : (i == 13)                         \
                                                                                      ? GPIO_PIN_REG_13             \
                                                                                      : (i == 14) ? GPIO_PIN_REG_14 \
                                                                                                  : GPIO_PIN_REG_15
```
strange style.

And you can compare to the original code.
But totally, clang-format is useful.

Reply:
Aligning macros is a feature that is coming (soon it seems) https://reviews.llvm.org/D28462. For headers with big sections of macros I guess reformatting can be disabled for now.

That last one is a big strange, I know. I can look more into the configuration details, but there is always the possibility to [ignore sections](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code).

I don't want to spend too much time on clang-format until that tools is the preferred tool, but I am happy to work on if it is selected. I've seen astyle mentioned too, someone might have a preference for that.